### PR TITLE
Add bootstrap cluster tasks

### PR DESCRIPTION
**Pull request by chrisdoherty4**
Design ref: https://github.com/aws/eks-anywhere/blob/main/designs/workflow-refactor.md

Add the bootstrap cluster create and delete tasks for the new management workflow. Tests for the tasks have been omitted intentionally as the context contracts between tasks are to be determined.

### CODE REVIEW

File: pkg/workflow/management/create.go [hunk 1]
```diff
@@ -15,26 +15,57 @@ type CreateClusterHookRegistrar interface {
...
+//  2. DeleteBootstrapCluster
+type CreateCluster struct {
+	// The spec used to construcft all other dependencies.
+	Spec *cluster.Spec
+
+	// CreateBootstrapOptions supplies bootstrap cluster options for creating bootstrap clusters.
+	CreateBootstrapOptions task.BootstrapOptionsRetriever
+
+	// Bootstrapper creates and destroys bootstrap clusters.
+	Bootstrapper *bootstrapper.Bootstrapper
```
Comment (g-gaston) [inline pkg/workflow/management/create.go:30]: should this be 
```suggestion
	Bootstrapper task.Bootstrapper
```

File: pkg/workflow/management/task/bootstrap.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+type Bootstrapper interface {
+	// CreateBootstrapCluster creates a new local cluster. It does not contain any EKS-A components.
+	CreateBootstrapCluster(
+		context.Context,
+		*cluster.Spec,
+		...bootstrapper.BootstrapClusterOption,
+	) (*types.Cluster, error)
+
+	// DeleteBootstrapCluster deletes a local cluster created with CreateBootstrapCluster.
+	DeleteBootstrapCluster(ctx context.Context, cluster *types.Cluster, isUpgrade bool) error
```
Comment (g-gaston) [inline pkg/workflow/management/task/bootstrap.go:30]: `isUpgrade` painful to see :(

**Review by jacobweinstock — APPROVED** [review]

**Pull request merged**